src/datasets/utils/file_utils.py

```diff
@@ -22,6 +22,7 @@
 from zipfile import ZipFile, is_zipfile
 
 import numpy as np
+import rarfile
```

Comment:
This import would make the library impossible to import for people that don't have `rarfile` installed.

To make it an optional import you can do the same as what we did for pytorch at the top of the `file_utils.py` file.

The idea is to have a boolean `_rarfile_available` that is True when `rarfile` is installed. Then before any call to `rarfile` you first have to check if it is avaibale using `_rarfile_available`.

Reply:
Thanks @lhoestq , I also added `rarfile` to `setup.py` file, does that make sense or I should remove it from there ?

Reply:
Refactored everything. I tested with and without `rarfile` and it seems to work, also tested on a dataset that I am adding with rar files. I also added an exception if the file is `rar` but the user doesn't have `rarfile` installed.

Reply:
Nice thanks !

And yes we should have rarfile in the `tests` requirements, thanks for adding it